mesonbuild/mesonlib.py

```diff
@@ -506,7 +506,6 @@ def detect_vcs(source_dir):
     return None
 
 # a helper class which implements the same version ordering as RPM
-@functools.total_ordering
```

Comment:
This works with `__eq__` and one other comparison operator, not `__cmp__`. Was it really faster (and less code) to re-implement all of the comparison?

Reply:
The benchmark time goes up from ~5 seconds to ~6.3 seconds if if use total_ordering with just `__lt__` and `__eq__`.

The original `__cmp__` implementation is 32 lines, 36 if we added the proper `if isinstance(other, Version) ... else: return NotImplemented` check. this comes out to 45 :shrug: